lib/json-schema/attribute.rb

```diff
@@ -33,8 +33,9 @@ def self.validation_errors(validator)
         'any' => Object,
       }
 
-      def self.data_valid_for_type?(data, type)
+      def self.data_valid_for_type?(data, type, nullable: false)
         valid_classes = TYPE_CLASS_MAPPINGS.fetch(type) { return true }
+        valid_classes = [valid_classes, NilClass] if nullable
```

Comment:
in case when type is boolean, you need to flatten the `valid_classes`

```rb
type = "boolean"
valid_classes = TYPE_CLASS_MAPPINGS.fetch(type) # => [TrueClass, FalseClass]
 valid_classes = [valid_classes, NilClass] if nullable # =>  [[TrueClass, FalseClass], NilClass]
 Array(valid_classes).any? { |c| data.is_a?(c) } # => #<TypeError: class or module required>
 ```
 
 


```

Reply:
Checking this @belinskidima